### PR TITLE
Change AttemptToFixRotation to only affect read in from DB

### PR DESCRIPTION
**Pull request by LtRipley36706**

### CODE REVIEW

**Comment by gmriggs** [comment]
I misunderstood https://github.com/ACEmulator/ACE/pull/3121, specifically where IsRotationValid() was being called

In the current implementation, IsRotationValid() is being called anytime .Location for any object on the server is being referenced. This is extremely bad design, and 1000% the wrong place / frequency to be calling IsRotationValid(). On a large server, this will result in 10s of thousands of calls to IsRotationValid() per second.

IsRotationValid() should be called when a position / rotation is first read in from the database

**Comment by Mag-nus** [comment]
Ripley, a good place to put the position check might be in WorldObject.cs, line 133, at the end of the WorldObject(Biota) ctor.

File: Source/ACE.Server/Entity/PositionExtensions.cs [hunk 1]
```diff
@@ -325,6 +325,9 @@ public static bool IsRotationValid(this Quaternion q)
 
         public static bool AttemptToFixRotation(this Position pos, WorldObject wo, PositionType positionType)
         {
+            if (wo is SpellProjectile)
```
Comment (gmriggs) [inline Source/ACE.Server/Entity/PositionExtensions.cs:328]: this bit shouldn't be needed now?

File: Source/ACE.Server/Entity/PositionExtensions.cs [hunk 2]
```diff
@@ -325,6 +325,9 @@ public static bool IsRotationValid(this Quaternion q)
...
+                return false;
+
```
Comment (LtRipley36706) [inline Source/ACE.Server/Entity/PositionExtensions.cs:330]:
```suggestion
```

**Pull request merged**